% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\PassOptionsToPackage{dvipsnames,svgnames,x11names}{xcolor}
%
\documentclass[
  11pt,
]{article}

\usepackage{amsmath,amssymb}
\usepackage{setspace}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else  
    % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\usepackage[margin=1.0in]{geometry}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
% Make \paragraph and \subparagraph free-standing
\makeatletter
\ifx\paragraph\undefined\else
  \let\oldparagraph\paragraph
  \renewcommand{\paragraph}{
    \@ifstar
      \xxxParagraphStar
      \xxxParagraphNoStar
  }
  \newcommand{\xxxParagraphStar}[1]{\oldparagraph*{#1}\mbox{}}
  \newcommand{\xxxParagraphNoStar}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
  \let\oldsubparagraph\subparagraph
  \renewcommand{\subparagraph}{
    \@ifstar
      \xxxSubParagraphStar
      \xxxSubParagraphNoStar
  }
  \newcommand{\xxxSubParagraphStar}[1]{\oldsubparagraph*{#1}\mbox{}}
  \newcommand{\xxxSubParagraphNoStar}[1]{\oldsubparagraph{#1}\mbox{}}
\fi
\makeatother


\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

\usepackage[left]{lineno}
\linenumbers
\modulolinenumbers
\usepackage{helvet}
\renewcommand*\familydefault{\sfdefault}
\usepackage[T1]{fontenc}
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\AtBeginDocument{%
\ifdefined\contentsname
  \renewcommand*\contentsname{Table of contents}
\else
  \newcommand\contentsname{Table of contents}
\fi
\ifdefined\listfigurename
  \renewcommand*\listfigurename{List of Figures}
\else
  \newcommand\listfigurename{List of Figures}
\fi
\ifdefined\listtablename
  \renewcommand*\listtablename{List of Tables}
\else
  \newcommand\listtablename{List of Tables}
\fi
\ifdefined\figurename
  \renewcommand*\figurename{Figure}
\else
  \newcommand\figurename{Figure}
\fi
\ifdefined\tablename
  \renewcommand*\tablename{Table}
\else
  \newcommand\tablename{Table}
\fi
}
\@ifpackageloaded{float}{}{\usepackage{float}}
\floatstyle{ruled}
\@ifundefined{c@chapter}{\newfloat{codelisting}{h}{lop}}{\newfloat{codelisting}{h}{lop}[chapter]}
\floatname{codelisting}{Listing}
\newcommand*\listoflistings{\listof{codelisting}{List of Listings}}
\makeatother
\makeatletter
\makeatother
\makeatletter
\@ifpackageloaded{caption}{}{\usepackage{caption}}
\@ifpackageloaded{subcaption}{}{\usepackage{subcaption}}
\makeatother

\ifLuaTeX
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage{bookmark}

\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same} % disable monospaced font for URLs
\hypersetup{
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}


\author{}
\date{}

\begin{document}


\setstretch{1.75}
\raggedright

\section{clustur: An R package for clustering features using sparse
distances
matrices}\label{clustur-an-r-package-for-clustering-features-using-sparse-distances-matrices}

\vspace{20mm}

\textbf{Running title:} clustur

\vspace{20mm}

Gregory Johnson Jr.\text, Sarah L. Westcott, Patrick D.
Schloss\textsuperscript{\textdagger}

\vspace{25mm}

\textdagger To whom correspondence should be addressed\\
\href{mailto:pschloss@umich.edu}{pschloss@umich.edu}

\vspace{10mm}

Department of Microbiology \& Immunology\\
University of Michigan\\
Ann Arbor, MI 48109

\vspace{20mm}

\textbf{Software Announcement}

\newpage

\subsection{Abstract}\label{abstract}

The clustur R package implements the \emph{de novo} clustering
algorithms found in the mothur software package for assigning 16S rRNA
gene sequences to operational taxonomic units (OTUs). Making these
algorithms accessible through the R ecosystem will foster their further
development, broader application, and integration within other R
packages.

\newpage

\subsection{Announcement}\label{announcement}

Taxonomic classification of 16S rRNA gene sequences has been a
persistent challenge in microbial ecology studies because reference
databases are incomplete (1). As an alternative, operational taxonomic
units (OTUs) have been widely used for describing and comparing
microbial communities. Although their biological interpretation is
controversial, OTUs are typically defined as a group of sequences that
are more than 97\% similar or less than 3\% dissimilar to each other
(2). Methods for applying that definition has resulted in a sizable
literature. Three general approaches have emerged for assigning
sequences to OTUs: \emph{de novo} clustering, closed reference
clustering or phylotyping, and open reference clustering (3--9). These
methods are available through popular packages including mothur and
QIIME2 (10, 11).

The clustur R package implements the \emph{de novo} clustering
algorithms implemented in mothur. The package name references its focus
on clustering and the names of its predecessors DOTUR and mothur (10,
12). This package was developed to help address two issues. First, users
would be able to more easily integrate the type of analysis that mothur
specializes in with popular analysis and visualization packages within
the R package ecosystem. Second, by making the code behind mothur's
clustering functions accessible through the R language, we hope to
encourage further development of the algorithms behind these functions
and analyses based on the output of the functions. The clustur package
implements hierarchical clustering algorithms including the furthest,
nearest, unweighted (i.e.~average), and weighted neighbor clustering
algorithms and the OptiClust algorithm. Functions implementing the
hierarchical algorithms already exist within R; however, their
implementations within clustur make use of a sparse input distance
matrix and output data for a single distance threshold. The benefits of
censoring distances larger than the threshold and only outputting data
for a single threshold include a smaller memory requirement and faster
execution times (4). clustur makes use of the Rcpp R package to
implement C++ code originally written for the mothur software package to
preserve the speed of the functions.

Users can install the clustur package via CRAN or through the devtools
package's install\_github function. The primary input to clustur's
functions is a sparse distance matrix and a count file. The sparse
distance matrix is a data.table package object with two columns
indicating the identifiers of the sequences being compared and a column
with the distance between those sequences; data for comparisons with a
distance larger than the desired threshold (e.g., 0.03) does not need to
be included. The count file is a data.table package object indicating
the number of times a sequence is found in each sample. The cluster
functions output two data.table objects. The first one has two columns
indicating the sequences and OTU identifiers. The second displays the
abundance of each sequence in each OTU. This has identical functionality
to the cluster and make.shared functions from mothur. Detailed vignettes
are available within the package to teach users how to install the
package, use its functions, and perform downstream analyses including
analysis within the vegan and ggplot2 R packages.

\subsection{Data availability}\label{data-availability}

clustur is available through CRAN and developmental versions are
available through the project's GitHub website
(https://github.com/schlosslab/clustur). The package is available under
the MIT open source license.

\subsection{Acknowledgements}\label{acknowledgements}

This project was supported, in part, by a grant from the US National
Institutes of Health (U01CA264071) to PDS.

\newpage

\subsection{References}\label{references}

\phantomsection\label{refs}
\begin{CSLReferences}{0}{1}
\bibitem[\citeproctext]{ref-Wang2007}
\CSLLeftMargin{1. }%
\CSLRightInline{\textbf{Wang Q}, \textbf{Garrity GM}, \textbf{Tiedje
JM}, \textbf{Cole JR}. 2007. Naïve bayesian classifier for rapid
assignment of rRNA sequences into the new bacterial taxonomy. Applied
and Environmental Microbiology \textbf{73}:5261--5267.
doi:\href{https://doi.org/10.1128/aem.00062-07}{10.1128/aem.00062-07}.}

\bibitem[\citeproctext]{ref-Stackebrandt1994}
\CSLLeftMargin{2. }%
\CSLRightInline{\textbf{STACKEBRANDT E}, \textbf{GOEBEL BM}. 1994.
Taxonomic note: A place for DNA-DNA reassociation and 16S rRNA sequence
analysis in the present species definition in bacteriology.
International Journal of Systematic and Evolutionary Microbiology
\textbf{44}:846--849.
doi:\href{https://doi.org/10.1099/00207713-44-4-846}{10.1099/00207713-44-4-846}.}

\bibitem[\citeproctext]{ref-NavasMolina2013}
\CSLLeftMargin{3. }%
\CSLRightInline{\textbf{Navas-Molina JA}, \textbf{Peralta-Sánchez JM},
\textbf{González A}, \textbf{McMurdie PJ}, \textbf{Vázquez-Baeza Y},
\textbf{Xu Z}, \textbf{Ursell LK}, \textbf{Lauber C}, \textbf{Zhou H},
\textbf{Song SJ}, \textbf{Huntley J}, \textbf{Ackermann GL},
\textbf{Berg-Lyons D}, \textbf{Holmes S}, \textbf{Caporaso JG},
\textbf{Knight R}. 2013.
\href{https://doi.org/10.1016/b978-0-12-407863-5.00019-8}{Advancing our
understanding of the human microbiome using QIIME}, p. 371--444.
\emph{In} Microbial metagenomics, metatranscriptomics, and
metaproteomics. Elsevier.}

\bibitem[\citeproctext]{ref-Schloss2011}
\CSLLeftMargin{4. }%
\CSLRightInline{\textbf{Schloss PD}, \textbf{Westcott SL}. 2011.
Assessing and improving methods used in operational taxonomic unit-based
approaches for 16S rRNA gene sequence analysis. Applied and
Environmental Microbiology \textbf{77}:3219--3226.
doi:\href{https://doi.org/10.1128/aem.02810-10}{10.1128/aem.02810-10}.}

\bibitem[\citeproctext]{ref-Schloss2016}
\CSLLeftMargin{5. }%
\CSLRightInline{\textbf{Schloss PD}. 2016. Application of a
database-independent approach to assess the quality of operational
taxonomic unit picking methods. mSystems \textbf{1}.
doi:\href{https://doi.org/10.1128/msystems.00027-16}{10.1128/msystems.00027-16}.}

\bibitem[\citeproctext]{ref-Westcott2015}
\CSLLeftMargin{6. }%
\CSLRightInline{\textbf{Westcott SL}, \textbf{Schloss PD}. 2015. De novo
clustering methods outperform reference-based methods for assigning 16S
rRNA gene sequences to operational taxonomic units. PeerJ
\textbf{3}:e1487.
doi:\href{https://doi.org/10.7717/peerj.1487}{10.7717/peerj.1487}.}

\bibitem[\citeproctext]{ref-Westcott2017}
\CSLLeftMargin{7. }%
\CSLRightInline{\textbf{Westcott SL}, \textbf{Schloss PD}. 2017.
OptiClust, an improved method for assigning amplicon-based sequence data
to operational taxonomic units. mSphere \textbf{2}.
doi:\href{https://doi.org/10.1128/mspheredirect.00073-17}{10.1128/mspheredirect.00073-17}.}

\bibitem[\citeproctext]{ref-Kopylova2016}
\CSLLeftMargin{8. }%
\CSLRightInline{\textbf{Kopylova E}, \textbf{Navas-Molina JA},
\textbf{Mercier C}, \textbf{Xu ZZ}, \textbf{Mahé F}, \textbf{He Y},
\textbf{Zhou H-W}, \textbf{Rognes T}, \textbf{Caporaso JG},
\textbf{Knight R}. 2016. Open-source sequence clustering methods improve
the state of the art. mSystems \textbf{1}.
doi:\href{https://doi.org/10.1128/msystems.00003-15}{10.1128/msystems.00003-15}.}

\bibitem[\citeproctext]{ref-He2015}
\CSLLeftMargin{9. }%
\CSLRightInline{\textbf{He Y}, \textbf{Caporaso JG}, \textbf{Jiang X-T},
\textbf{Sheng H-F}, \textbf{Huse SM}, \textbf{Rideout JR}, \textbf{Edgar
RC}, \textbf{Kopylova E}, \textbf{Walters WA}, \textbf{Knight R},
\textbf{Zhou H-W}. 2015. Stability of operational taxonomic units: An
important but neglected property for analyzing microbial diversity.
Microbiome \textbf{3}.
doi:\href{https://doi.org/10.1186/s40168-015-0081-x}{10.1186/s40168-015-0081-x}.}

\bibitem[\citeproctext]{ref-Schloss2009}
\CSLLeftMargin{10. }%
\CSLRightInline{\textbf{Schloss PD}, \textbf{Westcott SL},
\textbf{Ryabin T}, \textbf{Hall JR}, \textbf{Hartmann M},
\textbf{Hollister EB}, \textbf{Lesniewski RA}, \textbf{Oakley BB},
\textbf{Parks DH}, \textbf{Robinson CJ}, \textbf{Sahl JW}, \textbf{Stres
B}, \textbf{Thallinger GG}, \textbf{Van Horn DJ}, \textbf{Weber CF}.
2009. Introducing mothur: Open-source, platform-independent,
community-supported software for describing and comparing microbial
communities. Applied and Environmental Microbiology
\textbf{75}:7537--7541.
doi:\href{https://doi.org/10.1128/aem.01541-09}{10.1128/aem.01541-09}.}

\bibitem[\citeproctext]{ref-Bolyen2019}
\CSLLeftMargin{11. }%
\CSLRightInline{\textbf{Bolyen E}, \textbf{Rideout JR}, \textbf{Dillon
MR}, \textbf{Bokulich NA}, \textbf{Abnet CC}, \textbf{Al-Ghalith GA},
\textbf{Alexander H}, \textbf{Alm EJ}, \textbf{Arumugam M},
\textbf{Asnicar F}, \textbf{Bai Y}, \textbf{Bisanz JE},
\textbf{Bittinger K}, \textbf{Brejnrod A}, \textbf{Brislawn CJ},
\textbf{Brown CT}, \textbf{Callahan BJ}, \textbf{Caraballo-Rodríguez
AM}, \textbf{Chase J}, \textbf{Cope EK}, \textbf{Da Silva R},
\textbf{Diener C}, \textbf{Dorrestein PC}, \textbf{Douglas GM},
\textbf{Durall DM}, \textbf{Duvallet C}, \textbf{Edwardson CF},
\textbf{Ernst M}, \textbf{Estaki M}, \textbf{Fouquier J},
\textbf{Gauglitz JM}, \textbf{Gibbons SM}, \textbf{Gibson DL},
\textbf{Gonzalez A}, \textbf{Gorlick K}, \textbf{Guo J},
\textbf{Hillmann B}, \textbf{Holmes S}, \textbf{Holste H},
\textbf{Huttenhower C}, \textbf{Huttley GA}, \textbf{Janssen S},
\textbf{Jarmusch AK}, \textbf{Jiang L}, \textbf{Kaehler BD},
\textbf{Kang KB}, \textbf{Keefe CR}, \textbf{Keim P}, \textbf{Kelley
ST}, \textbf{Knights D}, \textbf{Koester I}, \textbf{Kosciolek T},
\textbf{Kreps J}, \textbf{Langille MGI}, \textbf{Lee J}, \textbf{Ley R},
\textbf{Liu Y-X}, \textbf{Loftfield E}, \textbf{Lozupone C},
\textbf{Maher M}, \textbf{Marotz C}, \textbf{Martin BD},
\textbf{McDonald D}, \textbf{McIver LJ}, \textbf{Melnik AV},
\textbf{Metcalf JL}, \textbf{Morgan SC}, \textbf{Morton JT},
\textbf{Naimey AT}, \textbf{Navas-Molina JA}, \textbf{Nothias LF},
\textbf{Orchanian SB}, \textbf{Pearson T}, \textbf{Peoples SL},
\textbf{Petras D}, \textbf{Preuss ML}, \textbf{Pruesse E},
\textbf{Rasmussen LB}, \textbf{Rivers A}, \textbf{Robeson MS},
\textbf{Rosenthal P}, \textbf{Segata N}, \textbf{Shaffer M},
\textbf{Shiffer A}, \textbf{Sinha R}, \textbf{Song SJ}, \textbf{Spear
JR}, \textbf{Swafford AD}, \textbf{Thompson LR}, \textbf{Torres PJ},
\textbf{Trinh P}, \textbf{Tripathi A}, \textbf{Turnbaugh PJ},
\textbf{Ul-Hasan S}, \textbf{Hooft JJJ van der}, \textbf{Vargas F},
\textbf{Vázquez-Baeza Y}, \textbf{Vogtmann E}, \textbf{Hippel M von},
\textbf{Walters W}, \textbf{Wan Y}, \textbf{Wang M}, \textbf{Warren J},
\textbf{Weber KC}, \textbf{Williamson CHD}, \textbf{Willis AD},
\textbf{Xu ZZ}, \textbf{Zaneveld JR}, \textbf{Zhang Y}, \textbf{Zhu Q},
\textbf{Knight R}, \textbf{Caporaso JG}. 2019. Reproducible,
interactive, scalable and extensible microbiome data science using QIIME
2. Nature Biotechnology \textbf{37}:852--857.
doi:\href{https://doi.org/10.1038/s41587-019-0209-9}{10.1038/s41587-019-0209-9}.}

\bibitem[\citeproctext]{ref-Schloss2005}
\CSLLeftMargin{12. }%
\CSLRightInline{\textbf{Schloss PD}, \textbf{Handelsman J}. 2005.
Introducing DOTUR, a computer program for defining operational taxonomic
units and estimating species richness. Applied and Environmental
Microbiology \textbf{71}:1501--1506.
doi:\href{https://doi.org/10.1128/aem.71.3.1501-1506.2005}{10.1128/aem.71.3.1501-1506.2005}.}

\end{CSLReferences}




\end{document}
