---
format:
  docx: default
  gfm: default
  pdf:
    documentclass: article
    keep-tex: true
    lof: false
    toc: false
    number-sections: false
bibliography: references.bib
csl: asm.csl
geometry: margin=1.0in
fontsize: 11pt
linestretch: 1.75
header-includes:
 - \usepackage[left]{lineno}
 - \linenumbers
 - \modulolinenumbers
 - \usepackage{helvet}
 - \renewcommand*\familydefault{\sfdefault}
 - \usepackage[T1]{fontenc}
execute:
  echo: FALSE
  tidy: TRUE
  eval: TRUE
  warning: FALSE
  message: FALSE
  cache: FALSE
---

\raggedright

# clustur: An R package for clustering features using sparse distances matrices 


\vspace{20mm}

**Running title:** clustur

\vspace{20mm}

Gregory Johnson Jr.\text, Sarah L. Westcott, Patrick D. Schloss^\textdagger^

\vspace{25mm}


\textdagger To whom correspondence should be addressed  
\href{mailto:pschloss@umich.edu}{pschloss@umich.edu}

\vspace{10mm}

Department of Microbiology & Immunology  
University of Michigan  
Ann Arbor, MI 48109  

\vspace{20mm}

**Software Announcement**

\newpage

## Abstract (needs to be under 50 words; currently 48)

The clustur R package implements the *de novo* clustering algorithms found in the mothur software package for assigning 16S rRNA gene sequences to operational taxonomic units (OTUs). Making these algorithms accessible through the R ecosystem will foster their further development, broader application, and integration within other R packages.


\newpage

## Announcement (needs to be ~500 words, currently 608)

Taxonomic classification of 16S rRNA gene sequences has been a persistent problem in microbial ecology studies because reference databases are incomplete [@Wang2007]. As an alternative, operational taxonomic units (OTUs) have been widely used for describing and comparing microbial communities. Although the biological interpretation is controversial, OTUs are typically defined as a group of sequences that are more than 97% similar or less than 3% dissimilar to each other [@Stackebrandt1994]. Methods for applying that definition has resulted in a sizable literature. Three general approaches have emerged for assigning sequences to OTUs. First, sequences can be clustered based on their similarity to each other. This approach has often been called "*de novo* clustering". Second, sequences can be clustered based on their similarity to reference sequences. This approach has often been called "closed reference clustering" or "phylotyping". Finally, a hybrid approach applies *de novo* clustering to sequences that are not sufficiently close to reference sequences when using closed reference clustering. This approach has been called "open reference clustering". We have described and compared these methods in great detail elsewhere [@Schloss2011; @Schloss2016; @Westcott2015; @Westcott2017]. These methods are available through popular packages including mothur and QIIME [@Schloss2009; @Bolyen2019].

The clustur R package implements the *de novo* clustering algorithms implemented in mothur. The package name references its focus on clustering and the names of its predecessors DOTUR and mothur [@Schloss2005; @Schloss2009]. This package was developed to help address two issues. First, users would be able to more easily integrate the type of analysis that mothur specializes in with popular analysis and visualization packages within the R package ecosystem. Second, by making the code behind mothur's clustering functions accessible through the R language, we hope to encourage further development of the algorithms behind these functions and analyses based on the output of the functions. The clustur package implements hierarchical clustering algorithms including the furthest, nearest, unweighted (i.e. average), and weighted neighbor clustering algorithms and the OptiClust algorithm. Although functions implementing the hierarchical algorithms already exist within R, their implementations within clustur make use of a sparse input distance matrix and output data for a single distance threshold. The benefits of censoring distances larger than the threshold and only outputting data for a single threshold include a smaller memory requirement and faster execution times. clustur makes use of the Rcpp R package to implement C++ code originally written for the mothur software package to preserve the speed of the functions.

Users can install the clustur package via CRAN or through the devtools package’s install_github function. The primary input to clustur’s functions is a sparse distance matrix and a count file. The sparse distance matrix is a data.table package object with two columns indicating the identifiers of the sequences being compared and a column with the distance between those sequences; data for comparisons with a distance larger than the desired threshold (e.g., 0.03) does not need to be included. The count file is a data.table package object indicating the number of times a sequence is found in each sample. The cluster functions output two data.table  objects.  The first one has two columns indicating the sequences and OTU identifiers . The second displays the abundance of each sequence in each OTU. This has identical functionality to the make.table command in mothur. Detailed vignettes are available within the package to teach users how to install the package, use its functions, and perform downstream analyses including analysis within the vegan and ggplot2 R packages.

## Data availability
clustur is available through CRAN and developmental versions are available through the project's GitHub website (https://github.com/schlosslab/clustur). The package is available under the MIT open source license.


## Acknowledgements
This project was supported, in part, by a grant from the US National Institutes of Health (U01CA264071) to PDS.


\newpage

## References

::: {#refs}
:::
